File: app/src/main/java/net/bloople/manga/audit/LibrariesAuditor.java
package net.bloople.manga.audit;

import android.content.Context;

import net.bloople.manga.Library;

public class LibrariesAuditor {
    private Context context;

    public LibrariesAuditor(Context context) {
        this.context = context;
    }

    public void selected(Library library) {
        AuditEvent event = new AuditEvent(
            System.currentTimeMillis(),
            Action.LIBRARY_SELECTED,
            ResourceType.UNKNOWN,
            AuditEvent.UNKNOWN_ID,
            ResourceType.LIBRARY,
            library.id(),
            library.name(),
            ""
        );
        event.save(context);
    }
}

File: app/src/main/java/net/bloople/manga/audit/LibrariesAuditor.kt
package net.bloople.manga.audit

import android.content.Context
import net.bloople.manga.Library

class LibrariesAuditor() {
    fun selected(library: Library) {
        val event = AuditEvent(
            System.currentTimeMillis(),
            Action.LIBRARY_SELECTED,
            ResourceType.UNKNOWN,
            AuditEvent.UNKNOWN_ID,
            ResourceType.LIBRARY,
            library.id,
            library.name!!,
            ""
        )
        event.save()
    }
}